app/src/main/java/net/gsantner/markor/frontend/filesearch/FileSearchEngine.java

```diff
@@ -97,7 +97,9 @@ public static FileSearchEngine.QueueSearchFilesTask queueFileSearch(Activity act
     public static class QueueSearchFilesTask extends AsyncTask<Void, Integer, List<FitFile>> {
         private final SearchOptions _config;
         private final GsCallback.a1<List<FitFile>> _callback;
-        private final Pattern _regex;
+
+        // _matcher.reset() is _not_ thread safe. Will need alternate approach when we make search parallel
+        private final Matcher _matcher;
```

Comment:
Does adding a synchronized(object) around the two reset calls solve the problem?

Reply:
There is no issue right now as this code is not parallel. In the future I think searching needs to be made faster and I just left this as a note for myself or another dev in the future.